app/controllers/procedures_controller.rb

```diff
@@ -57,6 +57,8 @@ def update
     @procedure.update_attributes(procedure_params)
     @appointment = @procedure.appointment
     @statuses = @appointment.appointment_statuses.map{|x| x.status}
+
+    @cost_error_message = (@procedure.errors.messages[:service_cost].select{|message| message == "No cost found, ensure that a valid pricing map exists for that date."}.first rescue nil)
```

Comment:
Since there aren't any specific validations of the `service_cost` attribute on the Procedure model, it's probably safe to say `@procedure.errors.messages[:service_cost].first` which will return nil if it's empty anyways. I would recommend this since string matching for the specific error seems kind of hacky.

Otherwise, `Enumerable#select` returns an array, so using `Enumerable#detect` would make a bit more sense here. It automatically finds the first item that causes the proc block to return true and if it doesn't find anything, already returns nil, so you wouldn't need to `rescue nil`.
`@procedure.errors.messages[:service_cost].detect{|message| message == "No cost found, ensure that a valid pricing map exists for that date."}`

Reply:
@kyle-glick Agreed on the .detect change. In terms of the specific string matching, I went back and forth, but decided to do it this way to not "ruin" any future validation that might crop up.